pennylane/plugins/default_gaussian.py

```diff
@@ -492,7 +492,7 @@ def set_state(state, wire, mu, cov):
 # ========================================================
 
 
-def photon_number(mu, cov, wires, params, total_wires, hbar=2.0):
+def photon_number(mu, cov, wires, params, total_wires, register, hbar=2.0):
```

Comment:
The entire `default_gaussian` plugin should be overhauled. The way it is done is that there is an operator map, and if an observable is called, PL maps it to a function and calls the function instead. 

Since different functions have different signatures, what was done here is to fake the signature of the longest function in all other functions. I just added another unused parameter, but there are many others.

Reply:
Ah, that's the historical context! :) I rewrote the code so that I only have to pass the ``register`` to the one function that actually uses it.